docs/pages/_app.js

```diff
@@ -189,11 +190,89 @@ function AppWrapper(props) {
     ];
   }
 
+  const { canonicalAs } = pathnameToLanguage(router.asPath);
+
   const pageContextValue = React.useMemo(() => {
     const { activePage, activePageParents } = findActivePage(pages, router.pathname);
 
-    return { activePage, activePageParents, pages };
-  }, [router.pathname]);
+    const languagePrefix = pageProps.userLanguage === 'en' ? '' : `/${pageProps.userLanguage}`;
+
+    let productIdentifier = null;
+
+    if (
+      canonicalAs.startsWith('/x/react-data-grid/') ||
+      canonicalAs.startsWith('/x/api/data-grid/')
+    ) {
+      productIdentifier = {
+        name: 'Data Grid',
+        metadata: 'MUI X',
+        versions: [
+          {
+            text: 'v6',
+            ...(process.env.DATA_GRID_VERSION.startsWith('6')
+              ? {
+                  text: `v${process.env.DATA_GRID_VERSION}`,
+                  current: true,
+                }
+              : {
+                  href: `https://mui.com${languagePrefix}/components/data-grid/`,
+                }),
+          },
+          {
+            text: 'v5',
+            ...(process.env.DATA_GRID_VERSION.startsWith('5')
+              ? {
+                  text: `v${process.env.DATA_GRID_VERSION}`,
+                  current: true,
+                }
+              : {
+                  href: `https://v5.mui.com${languagePrefix}/components/data-grid/`,
+                }),
+          },
+          { text: 'v4', href: `https://v4.mui.com${languagePrefix}/components/data-grid/` },
+        ],
+      };
+    } else if (
+      canonicalAs.startsWith('/x/react-date-pickers/') ||
+      canonicalAs.startsWith('/x/api/date-pickers/')
+    ) {
+      productIdentifier = {
+        name: 'Date pickers',
+        metadata: 'MUI X',
+        versions: [
+          {
+            ...(process.env.DATE_PICKERS_VERSION.startsWith('6')
+              ? {
+                  text: `v${process.env.DATE_PICKERS_VERSION}`,
+                  current: true,
+                }
+              : {
+                  text: `v6`,
+                  href: `https://next.mui.com${languagePrefix}/components/data-grid/`,
+                }),
+          },
+          {
+            ...(process.env.DATE_PICKERS_VERSION.startsWith('5')
+              ? {
+                  text: `v${process.env.DATE_PICKERS_VERSION}`,
+                  current: true,
+                }
+              : {
+                  text: `v5`,
+                  href: `https://v5.mui.com${languagePrefix}/components/data-grid/`,
+                }),
+          },
+        ],
+      };
+    } else if (canonicalAs.startsWith('/x/introduction/')) {
```

Comment:
What do you think about loosening this rule a bit so that anything else would get an `Advanced components` identifier instead of a blank space?
Stuff like the `migration` pages and the currently in progress charts... 🤔 
```suggestion
    } else if (canonicalAs.startsWith('/x/')) {
```